tests/test_pipeline/test_hierarchical_pipeline.py

```diff
@@ -0,0 +1,279 @@
+from unittest.mock import patch
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from etna.datasets import TSDataset
+from etna.metrics import MAE
+from etna.models import LinearPerSegmentModel
+from etna.models import NaiveModel
+from etna.pipeline.hierarchical_pipeline import HierarchicalPipeline
+from etna.reconciliation import BottomUpReconciliator
+from etna.reconciliation import TopDownReconciliator
+from etna.transforms import LinearTrendTransform
+from etna.transforms import MeanTransform
+
+
+@pytest.fixture
+def product_level_constant_hierarchical_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02", "2000-01-03"] * 4,
+            "segment": ["a"] * 3 + ["b"] * 3 + ["c"] * 3 + ["d"] * 3,
+            "target": [1, 1, 1] * 4,
```

Comment:
In the fixtures better to use different constants for different segments to make sure that pipeline does not mix them